charger/easee.go

```diff
@@ -509,7 +523,12 @@ func (c *Easee) waitForDynamicChargerCurrent(targetCurrent float64) error {
 			if value.(float64) == targetCurrent {
 				return nil
 			}
-		case <-timer.C: // time is up, bail
+		case <-timer.C: // time is up, bail after one final check
+			c.mux.Lock()
+			defer c.mux.Unlock()
+			if c.dynamicChargerCurrent == targetCurrent {
```

Comment:
This case is already covered remove-> remove

Reply:
I know. But as explained [above](https://github.com/evcc-io/evcc/pull/9018#issuecomment-1645404305) it happens more often than expected that a Product Update slips thought mutex.unlock and the start of the select. This catches these occurrences, and hence reduces the number of ERRORs we are generating.